walk-in-interview/src/main/webapp/script.js

```diff
@@ -122,98 +109,205 @@ function renderSelectOptions(id) {
 
-  /** reset the list so we don't render the same jobs twice */
+  /* reset the list so we don't render the same jobs twice */
   jobListingsElement.innerHTML = '';
+  jobShowing.innerHTML = '';
+
+  if (jobPageData === undefined ||
+    !jobPageData.hasOwnProperty('jobList') ||
+    jobPageData['jobList'].length === 0) {
+    setErrorMessage(/* msg= */ STRINGS['no-jobs-error-message'],
+        /* includesDefaultMsg= */ false);
+    return;
+  }
+
+  const jobListings = jobPageData['jobList'];
+
   const jobListingTemplate = document.getElementById('job-listing-template');
 
   jobListings.forEach((job) => {
     const jobListing =
-      jobListingTemplate.cloneNode( /** and child elements */ true);
+      jobListingTemplate.cloneNode( /* and child elements */ true);
     jobListing.setAttribute('id', job['jobId']);
 
     const jobTitle = jobListing.children[0];
     jobTitle.innerText = job['jobTitle'];
 
     const jobAddress = jobListing.children[1];
-
     const location = job['jobLocation'];
     jobAddress.innerText = `${location['address']}, ${location['postalCode']}`;
 
+    const jobPay = jobListing.children[2];
+    const pay = job['jobPay'];
+    jobPay.innerText = `${pay['min']} - ${pay['max']} SGD ` +
+      `(${pay['paymentFrequency'].toLowerCase()})`;
+
+    const requirementsList = jobListing.children[3];
+    const fullRequirementsList = getRequirementsList();
+    const requirementsArr = [];
+
+    job['requirements'].forEach((req) => {
+      requirementsArr.push(fullRequirementsList[req]);
+    });
+
+    requirementsList.innerText =
+      `Requirements List: ${requirementsArr.join(', ')}`;
+
     jobListingsElement.appendChild(jobListing);
   });
+
+  jobShowing.innerText = `${jobPageData['range'].minimum} -` +
+    ` ${jobPageData['range'].maximum} ${STRINGS['job-listings-showing']} ` +
+    `${jobPageData['totalCount']}`;
+}
+
+/**
+ * Add the list of jobs that are stored in the database
+ * given the filter fields.
+ */
+async function renderJobListings() {
+  if (!validateFilters()) {
+    return;
+  }
+
+  const sortingParam = document.getElementById('sort-by').value;
+
+  /*
+   * Note that this platform currently only sorts by Salary.
+   * TODO(issue/62): support more filters
+   */
+  if (!sortingParam.includes(SALARY_PARAM)) {
+    console.error('this app currently only sorts by salary');
+    return;
+  }
+
+  const sortByParam = SALARY_PARAM;
+  const orderByParam = sortingParam.substring(7);
+
+  const regionParam = document.getElementById('show-by-region').value;
+  let minLimitParam = document.getElementById('filter-min-limit')
+      .valueAsNumber;
+  let maxLimitParam = document.getElementById('filter-max-limit')
+      .valueAsNumber;
+
+  if (Number.isNaN(minLimitParam)) {
+    minLimitParam = 0;
+  }
+
+  if (Number.isNaN(maxLimitParam)) {
+    maxLimitParam = JAVA_INTEGER_MAX_VALUE;
+  }
+
+  const jobPageData = await getJobListings(regionParam, sortByParam,
```

Comment:
The function said async but here we do await which is sync?

Reply:
I thought the async/await keywords were used to make the code act more synchronously?